From 757886f37e898fc785e99275d0f60ffcd0d45a5c Mon Sep 17 00:00:00 2001
From: "renjiang.han" <renjiang.han@amlogic.com>
Date: Tue, 27 Apr 2021 15:08:43 +0800
Subject: [PATCH 6/8] amlvideo: fix amlvideo warn_on problem caused by
 querycap. [1/1]

PD#SWPL-47644

Problem:
amlvideo caused warn_on because of inconsistent results before
and after querycap.

Solution:
fix amlvideo warn_on problem caused by querycap.

Verify:
on AP222

Change-Id: I4329d71b8dc5b59707a4c12fda1b2aa1ae5b9ebe
Signed-off-by: renjiang.han <renjiang.han@amlogic.com>
---
 .../amlogic/media/video_processor/video_dev/amlvideo.c    | 8 +++-----
 1 file changed, 3 insertions(+), 5 deletions(-)

diff --git a/drivers/amlogic/media/video_processor/video_dev/amlvideo.c b/drivers/amlogic/media/video_processor/video_dev/amlvideo.c
index fab0143e2fac..67be5a7e650d 100644
--- a/drivers/amlogic/media/video_processor/video_dev/amlvideo.c
+++ b/drivers/amlogic/media/video_processor/video_dev/amlvideo.c
@@ -442,11 +442,9 @@ static int vidioc_querycap(struct file *file, void *priv,
 	strcpy(cap->card, "amlvideo");
 	strlcpy(cap->bus_info, dev->v4l2_dev.name, sizeof(cap->bus_info));
 	cap->version = AMLVIDEO_VERSION;
-	cap->device_caps = V4L2_CAP_VIDEO_CAPTURE;
-	cap->capabilities = cap->device_caps
-		| V4L2_CAP_DEVICE_CAPS
-		| V4L2_CAP_STREAMING
-		| V4L2_CAP_READWRITE;
+	cap->device_caps = V4L2_CAP_VIDEO_CAPTURE | V4L2_CAP_STREAMING
+			| V4L2_CAP_READWRITE;
+	cap->capabilities = cap->device_caps | V4L2_CAP_DEVICE_CAPS;
 	return 0;
 }
 
-- 
2.31.1

